webapp/javascript/components/Settings/Users/UserAddForm.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { useEffect, useState } from 'react';
+import Button from '@ui/Button';
+import Icon from '@ui/Icon';
+import { faTimes } from '@fortawesome/free-solid-svg-icons';
+import { formatRelative } from 'date-fns';
+import { request } from '@pyroscope/services/base';
+import { reloadUsers } from '@pyroscope/redux/reducers/settings';
+import { useAppDispatch } from '@pyroscope/redux/hooks';
+import { type User } from '../../../models/users';
+import styles from './UserForm.module.css';
+
+export type UserAddProps = User & { password?: string };
+
+function UserAddForm(props) {
+  const [form, setForm]: [UserAddProps, (value) => void] = useState({});
+  const dispatch = useAppDispatch();
+
+  const handleFormChange = (event) => {
+    const { name } = event.target;
+    const { value } = event.target;
+    setForm({ ...form, [name]: value });
+  };
+
+  const handleFormSubmit = (event) => {
+    console.log('Submitting user');
+    const data = {
+      ...form,
+      role: 'ReadOnly',
+      password: btoa(unescape(encodeURIComponent(form.password))),
+    };
+    console.log(JSON.stringify(data));
+    fetch('/api/users', {
+      method: 'POST',
+      body: JSON.stringify(data),
+      headers: [
+        ['Content-Type', 'application/json'],
+        ['Accept', 'application/json'],
+      ],
+    })
```

Comment:
I assume this is due to the PR being a WIP, but I think this should be moved to the `services/users`?

Reply:
Yeah, ofc all api related code goes to `services`. I just was at high speed :-)